### PR TITLE
SC-161-math-funcs

### PR DESCRIPTION
**Pull request by potan**
Add library

### CODE REVIEW

File: lang/jvm/src/main/scala/com/wavesplatform/lang/Global.scala [hunk 1]
```diff
@@ -32,4 +37,30 @@ object Global extends BaseGlobal {
...
+
+  // Math functions
+  def roundMode(round: BaseGlobal.Rounds) : RoundingMode = {
+    round match {
+      case BaseGlobal.RoundUp() => RoundingMode.UP
+      case BaseGlobal.RoundHalfUp() => RoundingMode.HALF_UP
+      case BaseGlobal.RoundDown() => RoundingMode.DOWN
+      case BaseGlobal.RoundHalfEven() => RoundingMode.HALF_EVEN
+      case BaseGlobal.RoundCeiling() => RoundingMode.CEILING
+      case BaseGlobal.RoundFloor() => RoundingMode.FLOOR
```
Comment (xrtm000) [inline lang/jvm/src/main/scala/com/wavesplatform/lang/Global.scala]: May be worth adding HALF_DOWN?

File: lang/shared/src/main/scala/com/wavesplatform/lang/v1/evaluator/ctx/impl/PureContext.scala [hunk 2]
```diff
@@ -505,30 +501,119 @@ object PureContext {
...
+          ("round", rounds, "round method")
+       ) {
+        case CONST_LONG(b) :: CONST_LONG(bp) :: CONST_LONG(e) :: CONST_LONG(ep) :: CONST_LONG(rp) :: round :: Nil =>
+          if(
+               bp < 0
+            || bp > 8
+            || ep < 0
+            || ep > 8
+            || rp < 0
+            || rp < 0
```
Comment (xrtm000) [inline lang/shared/src/main/scala/com/wavesplatform/lang/v1/evaluator/ctx/impl/PureContext.scala]: Duplicate condition

File: lang/shared/src/main/scala/com/wavesplatform/lang/v1/evaluator/ctx/impl/PureContext.scala [hunk 3]
```diff
@@ -505,30 +501,119 @@ object PureContext {
...
+  val roundHalfDown = CASETYPEREF("HalfDown", List.empty)
+  val rounds = UNION(roundDown, roundUp, roundHalfUp, roundHalfDown, roundCeiling, roundFloor, roundHalfEven)
+
+  def roundMode(m: EVALUATED): BaseGlobal.Rounds = {
+    m match {
+      case (p: CaseObj) => p.caseType.name match {
+        case "Down" => BaseGlobal.RoundDown()
+        case "Up" => BaseGlobal.RoundUp()
+        case "HalfUp" => BaseGlobal.RoundHalfUp()
+        case "HalfEven" => BaseGlobal.RoundHalfEven()
```
Comment (mrkraft) [inline lang/shared/src/main/scala/com/wavesplatform/lang/v1/evaluator/ctx/impl/PureContext.scala]: "HalfUp"

File: lang/shared/src/main/scala/com/wavesplatform/lang/v1/evaluator/ctx/impl/PureContext.scala [hunk 4]
```diff
@@ -505,30 +501,119 @@ object PureContext {
...
     Seq(
       UNIT,
       LONG,
       BOOLEAN,
       BYTESTR,
-      STRING
+      STRING,
+      roundDown,
+      roundUp,
+      roundHalfUp,
```
Comment (mrkraft) [inline lang/shared/src/main/scala/com/wavesplatform/lang/v1/evaluator/ctx/impl/PureContext.scala]: roundHalfDown

File: lang/shared/src/main/scala/com/wavesplatform/lang/v1/evaluator/ctx/impl/PureContext.scala (hunk 3 above)
Comment (xrtm000) [reply lang/shared/src/main/scala/com/wavesplatform/lang/v1/evaluator/ctx/impl/PureContext.scala]: I guess "HalfDown"

**Pull request merged**